tmt/beakerlib.py

```diff
@@ -182,6 +214,28 @@ def _library_cache(self) -> Dict[str, 'Library']:
 
     def fetch(self) -> None:
         """ Fetch the library (unless already fetched) """
+        if self.format == 'file':
```

Comment:
This does not seem right, it's piggybacking file/directory requirements on top of beakerlib class. The block below does not seem to use any beakerlib-specific functionality, that's unexpected. There's a shared concept, but I'm not sure it's wise to "hide" the pure file/directory requirement into a class dedicated to something more complex.

I'm sorry but it feels shoehorned into this spot. It's a second requirement more complex than a simple package name, and we should probably take care of creating it properly, as it's no longer just packages and beakerlibs. It deserves its own clear separation from the rest. Maybe beakerlib can build upon this new requirement type, maybe not, but they do not seem to be the same. For example, `Library` as a class is misused for both beakerlib libraries and mere directory copying, like a diversion. Maybe we need a distinct type for this, to implement not-a-package requirement materialization, with the current `Library` becoming its subclass, `BeakerlibLibrary`, and we could move this functionality into another subclass. There can have a `fetch()` method with nicely documented semantics, and whatever magic is needed for `file` won't bother `library`, and vice versa. I guess there is some overlap but isn't this too much pushed into *Beakerlib* library fetching and processing?

Reply:
refactored into separate package and modules